codemate/validator.py

```diff
@@ -0,0 +1,57 @@
+import ast
+
+from codemate.exceptions import InputError
+
+ERROR_CONTEXT_PADDING = 2
+ERROR_POST_FIX = "   <-- 🔍 seems the error is around here"
+
+
+def _format_error(syntax: str, error: SyntaxError) -> InputError:
+    if error.lineno is None:
+        raise ValueError("SyntaxError.lineno can not contain None")
+    lines = syntax.split("\n")
+    start = max(error.lineno - ERROR_CONTEXT_PADDING - 1, 0)
+    end = min(error.lineno + ERROR_CONTEXT_PADDING, len(lines))
+    for i, line_number in enumerate(range(start, end), start=start):
+        lines[i] = f"{str(line_number + 1).ljust(len(str(end)) + 1, ' ')}|{lines[i]}"
+    exc_index = error.lineno - 1
+    lines[exc_index] = lines[exc_index] + ERROR_POST_FIX
+    err_lines = "\n".join(lines[start:end])
+    exc_name = getattr(type(error), "__name__", "SyntaxError")
+    return InputError(
```

Comment:
Think as for `errors` they should always be raised not in return statements.

Reply:
In this function we are creating the error object, in the upper function we raise it